hibernate-core/src/test/java/org/hibernate/orm/test/metamodel/genericmodel/GenericMappedSuperclassMetamodelTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.orm.test.metamodel.genericmodel;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.MappedSuperclass;
+import jakarta.persistence.metamodel.EntityType;
+import org.hibernate.testing.orm.junit.EntityManagerFactoryScope;
+import org.hibernate.testing.orm.junit.Jpa;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Yanming Zhou
+ */
+@Jpa(
+		annotatedClasses = {
+				GenericMappedSuperclassMetamodelTest.Book.class,
+				GenericMappedSuperclassMetamodelTest.Owner.class,
+				GenericMappedSuperclassMetamodelTest.OwnerContainer.class
+		}
+)
+public class GenericMappedSuperclassMetamodelTest {
+
+	@Test
+	public void testGenericAttributeFromMappedSuperclassIsNotTypeErased(EntityManagerFactoryScope scope) {
+		scope.inTransaction( entityManager -> {
+			EntityType<Book> bookType = entityManager.getMetamodel().entity( Book.class );
+			assertEquals( Owner.class, bookType.getAttribute("owner").getJavaType() );
+		});
```

Comment:
I don't think this assertion is correct. As with Java's reflection API, `EntityType.getAttribute()` gives you a representation of the inherited attribute _as it was declared in the supertype_. It does _not_ do substitution of generic type arguments from the inheriting subclass, and I don't see any support anywhere in the JPA API or specification which hints that it should.

So I believe that it's correct for Hibernate to return `Object.class` here, and I think this PR should be closed.